src/fcntl.rs

```diff
@@ -164,6 +167,26 @@ pub fn openat<P: ?Sized + NixPath>(dirfd: RawFd, path: &P, oflag: OFlag, mode: M
     Errno::result(fd)
 }
 
+pub fn rename<P1: ?Sized + NixPath, P2: ?Sized + NixPath>(old: &P1, new: &P2) -> Result<()> {
```

Comment:
Is there any reason to use `rename` when `renameat` is supported?  None that I know of.

Reply:
None that I know of either. I added it because I saw in other cases (such as `open` and `openat` directly above) both variants were there. Would you like me to remove it?

Reply:
Yeah, you should remove it.  Supporting legacy APIs in existing code makes sense, but much like CloudABI, we shouldn't add brand new APIs that are already legacy.

Reply:
Done, and squashed my commits.